esp32/mods/modbt.c

```diff
@@ -472,22 +458,6 @@ static void create_hash(uint32_t pin, uint8_t *h_value)
     mbedtls_sha1_free(&sha1_context);
 }
 
-static bool is_pin_valid(uint32_t pin)
-{
-    int digits = 0;
-
-    while(pin != 0)
-    {
-        digits++;
-        pin /= 10;
-    }
- 
-    if (digits != MOD_BT_PIN_LENGTH){
-       return false;
-    }
-
-    return true;
-}
```

Comment:
It is a problem that length of `pin` is not inspected anymore. If pin can start with `0`, and we store it in an integer, we cannot check the length. I have tested the commit and expreienced the following:

Initiated instance with `bluetooth = Bluetooth(secure=True, pin=1234)` on LoPy4, without getting any error (intended because of the deleted code snippet), and tried to bond from a phone, using nRF Connect. Using `1234` pin code, connection cannot be created, instead, with `001234`, it was OK.

There was no other issue with any 6 digit pin codes (i.e. `050399` or `000001`), and pin longer than 6 digits still gives an error (i.e. `1234567`).

Do we want to:

1. Ignore pins starting with zero (official version)
2. Handle pins starting with zeros, but doesn't check pin minimal length? (this commit)
3. Handle both length, and enable pin starting with zero? (new implementation, I think we cannot check length, as we represent pin as int -> `001` == `000001`).

Reply:
This is intentional and as expected. As a direct result of the BLE implementation using an integer type to represent the passkey the leading zero's may be omitted in the calling Python code, but the resulting passkey will always be 6 digits. From the Bluetooth Core specification:
> [...] the Bluetooth passkey is a 6-digit numerical value. It is represented as integer value in the range 0x00000000 – 0x000F423F (000000 to 999999).

Note that this is also documented in the additional documentation for the user-facing `set_pin()` method.